Cargo.toml

```diff
@@ -4,57 +4,47 @@ version = "0.1.0"
 edition = "2021"
 license = "MIT OR Apache-2.0"
 
+# Ensure all dependencies + feats are mapped to crate features for correct usage
+# default features often have std breaking no_std and potentially other unwanted
 [dependencies]
-aead = "0.5.2"
-aes-gcm = "0.10.3"
-block-buffer = "0.10.4"
-chacha20poly1305 = "0.10.1"
-crypto-common = "0.1.6"
-der = "0.7.8"
-digest = "0.10.7"
-ecdsa = "0.16.8"
-ed25519-dalek = { version = "2.0.0", features = ["pkcs8"] }
-elliptic-curve = "0.13.5"
-generic-array = "0.14.7"
-hmac = "0.12.1"
-p256 = { version = "0.13.2", features = ["ecdh"] }
-p384 = { version = "0.13.0", features = ["ecdh"] }
-paste = "1.0.14"
-pkcs1 = { version = "0.7.5", features = ["pem", "pkcs8"] }
-pkcs8 = { version = "0.10.2", features = ["pem", "pkcs5"] }
+aead = { version = "0.5.2", default-features = false }
+aes-gcm = { version = "0.10.3", default-features = false, features = ["aes", "alloc"] }
+chacha20poly1305 = { version = "0.10.1", default-features = false }
+crypto-common = { version = "0.1.6", default-features = false }
+der = { version = "0.7.8", default-features = false }
+digest = { version = "0.10.7", default-features = false }
+ecdsa = { version = "0.16.8", default-features = false, features = ["alloc"] }
+ed25519-dalek = { version = "2.0.0", default-features = false, features = ["pkcs8"] }
+hmac = { version = "0.12.1", default-features = false }
+# TODO: figure out why std was needed here
+p256 = { version = "0.13.2", default-features = false, features = ["pem", "ecdsa", "ecdh"] }
+p384 = { version = "0.13.0", default-features = false, features = ["pem", "ecdsa", "ecdh"] }
+paste = { version = "1.0.14", default-features = false }
+pkcs8 = { version = "0.10.2", default-features = false, features = ["pem", "pkcs5"] }
 pki-types = { package = "rustls-pki-types", version = "1.0.1", default-features = false }
-rand_core = "0.6.4"
-rsa = { version = "0.9.2", features = ["sha2"] }
+rand_core = { version = "0.6.4", default-features = false, features = ["getrandom"] }
+rsa = { version = "0.9.2", default-features = false, features = ["sha2"] }
 rustls = { version = "0.23.0", default-features = false }
-sec1 = { version = "0.7.3", features = ["pkcs8", "pem"] }
-sha2 = "0.10.7"
-signature = "2.1.0"
-subtle = "2.5.0"
-typenum = "1.16.0"
+sec1 = { version = "0.7.3", default-features = false, features = ["pkcs8", "pem"] }
+sha2 = { version = "0.10.7", default-features = false }
+signature = { version = "2.1.0", default-features = false }
 webpki = { package = "rustls-webpki", version = "0.102.0", default-features = false }
-x25519-dalek = "2"
+x25519-dalek = { version = "2.0", default-features = false }
```

Comment:
`2` and `2.0` are equivalent, except dependabot will try to force upgrades for the explicitly specified part, which we could potentially merge although it may be overly restrictive

Reply:
e.g. say if we say "2.0" here for dependency it will not pick up 2.1 
 ... the same way if we say "2" it will neither pick up 2.1 ? was this correct ?

Reply:
Raised issue to capture this
- #44

Reply:
No, `2`, `2.0`, and `2.0.0` are completely equivalent. We tried to use upper version bounds in some cases, but they were more trouble than worth. For MSRV and outside dependencies we would have to wait for MSRV-aware resolver.

Reply:
I know the bounds are terrible agreed - but I don't remember 2.0.0 resolving to 2.1.0 automatically ?

Reply:
https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html

Reply:
Like this: https://github.com/pinkforest/test-semver-minor-bumps built a test case - the resolver is okay with X.Y.Z but the bounds otherwise usually has tricky behaviour in the resolver which leads up people writing as x.y.z to control the minor bumps ?

Reply:
Yeah we are not specifying bounds at all so saying "4.0.0" is by default ^4.0.0 resolving to 4.0.X only

See proof here: https://github.com/pinkforest/test-semver-minor-bumps

Reply:
Oh wait it was just me reading wrong.. .sigh :woman_facepalming: :woman_facepalming: I proved myself wrong

Reply:
Fixed - sorry about my epic confusion :sweat_smile: